packages/eas-cli/src/metadata/apple/config/writer.ts

```diff
@@ -64,15 +64,44 @@ export class AppleConfigWriter {
     };
   }
 
-  public setCategories({ primaryCategory, secondaryCategory }: AttributesOf<AppInfo>): void {
+  public setCategories(
+    attributes: Pick<
+      AttributesOf<AppInfo>,
+      | 'primaryCategory'
+      | 'primarySubcategoryOne'
+      | 'primarySubcategoryTwo'
+      | 'secondaryCategory'
+      | 'secondarySubcategoryOne'
+      | 'secondarySubcategoryTwo'
+    >
+  ): void {
+    this.schema.categories = undefined;
+    if (!attributes.primaryCategory && !attributes.secondaryCategory) {
+      return;
+    }
+
     this.schema.categories = [];
 
-    // TODO: see why these types are conflicting
-    if (primaryCategory) {
-      this.schema.categories.push(primaryCategory.id as any);
-      if (secondaryCategory) {
-        this.schema.categories.push(secondaryCategory.id as any);
-      }
+    if (attributes.primaryCategory && attributes.primarySubcategoryOne) {
+      this.schema.categories[0] = [
+        attributes.primaryCategory.id,
+        attributes.primarySubcategoryOne?.id,
+        attributes.primarySubcategoryTwo?.id,
+      ].filter(Boolean) as string[];
+    } else {
+      // If only the secondaryCategory was provided,
+      // autofill with an empty string and cause a store config error.
+      this.schema.categories[0] = attributes.primaryCategory?.id ?? '';
+    }
+
+    if (attributes.secondaryCategory && attributes.secondarySubcategoryOne) {
+      this.schema.categories[1] = [
+        attributes.secondaryCategory.id,
+        attributes.secondarySubcategoryOne?.id,
+        attributes.secondarySubcategoryTwo?.id,
+      ].filter(Boolean) as string[];
```

Comment:
1. I guess you're trying to filter falsy values here. Unfortunately, filtering like this is not correct. See:
  <img width="159" alt="Screenshot 2022-07-04 at 09 15 54" src="https://user-images.githubusercontent.com/5256730/177102075-cba042c6-04f6-4340-aee5-4bbdc06938d0.png">
2. It should be sth like:

```
const arr = [
    undefined,
    'string',
    null,
]

const isString = (value: string | undefined | null): value is string => typeof value == 'string';
const filtered = arr.filter(isString)
```